dbt_coves/tasks/generate/sources.py

```diff
@@ -253,30 +323,75 @@ def get_nested_keys(self, columns, schema, relation):
                     )
         return result
 
-    def render_templates(self, relation, columns, destination, nested=None):
+    def render_templates(
+        self, relation, columns, destination, nested=None, meta_d=None
+    ):
         context = {
             "relation": relation,
             "columns": columns,
             "nested": {},
             "adapter_name": self.adapter.__class__.__name__,
         }
+
         if nested:
-            context["nested"] = self.get_nested_keys(
-                nested, relation.schema, relation.name
-            )
-            # Removing original column with JSON data
-            new_cols = []
-            for col in columns:
-                if col.name.lower() not in context["nested"]:
-                    new_cols.append(col)
-            context["columns"] = new_cols
+            if meta_d:
+                nested_columns = self.get_nested_keys(
+                    nested, relation.schema, relation.name
+                )
+                new_cols = []
+                for col in columns:
+                    if col.column.lower() not in nested_columns:
+                        nc = {}
+                        nc["name"] = col.column.lower()
+                        for x in meta_d.values():
+                            if nc["name"] == x["name"].lower():
+                                nc["dtype"] = x["type"].lower()
+                                nc["description"] = x["description"]
+                                nc["col_instance"] = col
+                                new_cols.append(nc)
+                context["columns"] = new_cols
+                new_nes = {}
+                for reg, cols in nested_columns.items():
+                    for col in cols:
+                        for x in meta_d.values():
+                            if col.lower() in x["name"].lower():
```

Comment:
Again, you not using the keys defined. The whole idea of those keys was just to avoid nested fors in the code.